structuralalignment/superposition/theseus.py

```diff
@@ -152,49 +105,66 @@ def _filemap(self):
         """
         Create filemap for Theseus
         """
-        filenames = self.fetched_pdbs_filename
         with open(self.filemap_file, "w") as outfile:
-            for fname in filenames:
+            for fname in self.pdbs_filename:
                 outfile.write(f"{fname} {fname}\n")
 
-    def _calculate(self, structures, **kwargs):
+    def _calculate(self, structures, identical: bool = True, **kwargs):
         """
         Align the sequences with an alignment tool (``muscle``)
 
         .. todo::
 
             Can we use a different alignment tool, preferrably in Python? E.g. biotite?
 
-        Parameters
-        ----------
+
+
+        A mode for superpositioning macromolecules with
+        identical sequences and numbers of residues,
+        for instance, multiple models in an NMR family
+        or multiple structures from different crystal
+        forms of the same protein.
+
+
+        Parameter
+        ---------
         structures : list
             list of atomium objects
+
+        **kwargs : dictionary
+            optional parameter
         """
+
         with enter_temp_directory(remove=False) as (cwd, tmpdir):
             print("DEBUG: Running in", tmpdir, " -- TODO: DELETE BEFORE MERGE --")
-            self._create_pdb(structures)
-            self._get_fasta()
-            self._concatenate_fasta()
-            self._filemap()
-            seq_alignment_output = subprocess.check_output(
-                [
-                    self.alignment_app,
-                    "-maxiters",
-                    str(self.max_iterations),
-                    "-in",
-                    self.fastafile,
-                    "-out",
-                    self.alignment_file,
-                    "-clwstrict",
-                ],
-                universal_newlines=True,
-            )
-            self._parse_alignment(seq_alignment_output)
-            print(seq_alignment_output)
-            superposition_output = self._run_theseus()
-            print(superposition_output)
-            print("-- OUTPUT WE NEED TO PARSE BEFORE DELETING TEMPFILES --")
-            print(*list(Path(tmpdir).glob("theseus_*")), sep="\n")
+
+            self.pdbs_filename = self.__create_pdb(structures)
```

Comment:
I don't think we need to keep state here. `pdbs_filename` is a temporary list of temporary filenames we won't be able to reach later anyway, so there's no use in keeping that within `self`.

Reply:
I did that so I can use the filenames in other methods too and didn't want to call self.__create_pdb() everytime

Reply:
Thing is these files won't survive the execution of `._calculate()`, so keeping them in `self` is a recipe for tricky bugs. If you need them in other methods, pass them explicitly using arguments. The factory class should only store configuration values - the idea is that we can reuse it with no modifications for several structures.